invenio_accounts/models.py

```diff
@@ -1,235 +1,74 @@
-            if field in data and getattr(self, field) != data[field]:
-                changed_attrs[field] = getattr(self, field)
-                setattr(self, field, data[field])
-
-        if 'email' in changed_attrs:
-            self.verify_email(force=True)
-
-        try:
-            db.session.commit()
-        except Exception:
-            db.session.rollback()
-            raise
-        current_user.reload()
-        profile_updated.send(
-            sender=self.id, user=self, changed_attrs=changed_attrs
-        )
-
-        return changed_attrs
-
-    @property
-    def guest(self):
-        """Return True if the user is a guest."""
-        return False if self.email else True
-
-    #
-    # Basic functions for user authentification.
-    #
-    def get_id(self):
-        """Return the id."""
-        return self.id
-
-    def is_confirmed(self):
-        """Return true if accounts has been confirmed."""
-        return self.note == "1"
-
-    def is_guest(self):
-        """Return if the user is a guest."""
-        return self.guest
-
-    def is_authenticated(self):
-        """Return True if user is a authenticated user."""
-        return True if self.email else False
-
-    def is_active(self):
-        """Return True if use is active."""
-        return self.note != "0"
-
-
-class UserEXT(db.Model):
-
-    """Represent a UserEXT record."""
-
-    __tablename__ = 'userEXT'
-
-    id = db.Column(db.String(255), primary_key=True, nullable=False)
-    method = db.Column(db.String(50), primary_key=True, nullable=False)
-    id_user = db.Column(db.Integer(15, unsigned=True),
-                        db.ForeignKey(User.id), nullable=False)
-
-    user = db.relationship(User, backref="external_identifiers")
-
-    __table_args__ = (
-        db.Index('userext_id_user_method', id_user, method, unique=True),
-        db.Model.__table_args__
-    )
-
-__all__ = ('User', 'UserEXT')
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Database models for accounts."""
+
+from __future__ import absolute_import, print_function
+
+from flask_security import RoleMixin, UserMixin
+from invenio_db import db
+from sqlalchemy_utils import IPAddressType
+
+roles_users = db.Table(
+    'roles_users',
+    db.Column('user_id', db.Integer(), db.ForeignKey('user.id')),
+    db.Column('role_id', db.Integer(), db.ForeignKey('role.id')),
+)
+
+
+class Role(db.Model, RoleMixin):
+    """Role data model."""
+
+    id = db.Column(db.Integer(), primary_key=True)
+
+    name = db.Column(db.String(80), unique=True)
+
+    description = db.Column(db.String(255))
+
+
+class User(db.Model, UserMixin):
+    """User data model."""
+
```

Comment:
Should we go with a naming scheme: ``accounts_role``? or will we hit the mysql 16 characters limit too fast?